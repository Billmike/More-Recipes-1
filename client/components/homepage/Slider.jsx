import React from 'react';

import slide1 from '../../public/images/slide1.jpg';
import slide2 from '../../public/images/slide2.jpg';
import slide3 from '../../public/images/slide3.jpg';
import slide4 from '../../public/images/slide4.jpg';

/**
 * Class representing slider
 *
 * @class Slider
 *
 * @extends {React.Component}
 */
class Slider extends React.Component {
  /**
   * Component did mount method
   *
   * @returns {function} Initialize the slider
   *
   * @memberof Slider
   */
  componentDidMount() {
    $('.slider').slider({
      full_width: true,
      height: 600,
    });
  }
  /**
   * Render method
   *
   * @returns {object} React element
   *
   * @memberof Slider
   */
  render() {
    return (
      <div className="slider col s12">
        <ul className="slides">
          <li>
            <img src={slide1} alt="Picture1" />
            <div className="caption center-align">
              <h3>The taste of nature&apos;s gift.</h3>
              <h5 className="light white-text">
                The source of healthy living.
              </h5>
            </div>
          </li>
          <li>
            <img src={slide2} alt="Picture2" />
            <div className="caption left-align">
              <h3>We eat to live.</h3>
              <h5 className="light white-text text-lighten-3">
                Let the food be worth it.
              </h5>
            </div>
          </li>
          <li>
            <img src={slide3} alt="Picture3" />
            <div className="caption right-align">
              <h3>Let healthiness be your priority.</h3>
              <h5 className="light white-text text-lighten-3">
                Health is wealth.
              </h5>
            </div>
          </li>
          <li>
            <img src={slide4} alt="Picture4" />
            <div className="caption center-align">
              <h3>Share your wonderful recipe</h3>
              <h5 className="light white-text text-lighten-3">
                Never let your ideas get lost.
              </h5>
            </div>
          </li>
        </ul>
      </div>
    );
  }
}

export default Slider;
