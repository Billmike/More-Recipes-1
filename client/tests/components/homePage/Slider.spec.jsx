import React from 'react';

import Slider from '../../../components/homepage/Slider';

describe('Slider component', () => {
  let spy;

  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    spy = jest.spyOn(Slider.prototype, 'componentDidMount');
    const wrapper = mount(<Slider />);

    expect(wrapper).toBeDefined();
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
