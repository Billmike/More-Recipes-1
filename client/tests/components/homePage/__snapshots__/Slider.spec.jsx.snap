// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component should render component 1`] = `
<Slider>
  <div
    className="slider col s12"
  >
    <ul
      className="slides"
    >
      <li>
        <img
          alt="Picture1"
          src="test-file-stub"
        />
        <div
          className="caption center-align"
        >
          <h3>
            The taste of nature's gift.
          </h3>
          <h5
            className="light white-text"
          >
            The source of healthy living.
          </h5>
        </div>
      </li>
      <li>
        <img
          alt="Picture2"
          src="test-file-stub"
        />
        <div
          className="caption left-align"
        >
          <h3>
            We eat to live.
          </h3>
          <h5
            className="light white-text text-lighten-3"
          >
            Let the food be worth it.
          </h5>
        </div>
      </li>
      <li>
        <img
          alt="Picture3"
          src="test-file-stub"
        />
        <div
          className="caption right-align"
        >
          <h3>
            Let healthiness be your priority.
          </h3>
          <h5
            className="light white-text text-lighten-3"
          >
            Health is wealth.
          </h5>
        </div>
      </li>
      <li>
        <img
          alt="Picture4"
          src="test-file-stub"
        />
        <div
          className="caption center-align"
        >
          <h3>
            Share your wonderful recipe
          </h3>
          <h5
            className="light white-text text-lighten-3"
          >
            Never let your ideas get lost.
          </h5>
        </div>
      </li>
    </ul>
  </div>
</Slider>
`;
